x-pack/test/search_sessions_integration/tests/apps/discover/async_search.ts

```diff
@@ -128,6 +131,19 @@ export default function ({ getPageObjects, getService }: FtrProviderContext) {
 
       expect(searchesCountBeforeRestore).to.be(searchesCountAfterRestore); // no new searches started during restore
     });
+
+    it('should should clean the search session when navigating to SQL mode, and reinitialize when navigating back', async () => {
```

Comment:
I accidentally broke this while working on this fix, so decided to add a test for it.